frontend/app/src/components/asset-manager/ManagedAssetForm.vue

```diff
@@ -97,10 +116,16 @@ watch(address, async () => {
     name: newName,
     symbol: newSymbol
   } = await fetchTokenDetails(get(address));
-  set(decimals, newDecimals?.toString() || get(decimals));
+  set(decimals, newDecimals ?? get(decimals));
   set(name, newName || get(name));
   set(symbol, newSymbol || get(symbol));
   set(fetching, false);
+  clearFieldErrors(['decimals', 'name', 'symbol']);
+});
+
+watch(assetType, () => {
+  // clearing these errors because these fields are disabled for non evm assets
+  clearFieldErrors(['evmChain', 'tokenKind']);
```

Comment:
Out of curiosity why not clear all the errors when changing asset type?

Since each asset type has a different set of required fields we should be able to assume that any previous validation error is irrelevant.

Reply:
okay, that'll do...

Reply:
running tests, to be sure it's all good.